### PR TITLE
resource_control: publish runaway management rfc

### PR DESCRIPTION
**Pull request by Connor1996**
### What problem does this PR solve?

Issue Number: Ref #43691 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: docs/design/2023-06-16-runaway-queries-management.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+- SQL_TEXT: text of the runaway sql
+- SQL_DIGEST: plan digest of the runaway sql
+
+#### Option1: persist in log file
+
+Runaway query records are stored in a dedicated, auto rotated, run-away log file, quite like slow log file. And the records can be also viewed by the admin table INFORMATION_SCHEMA.RUNAWAY_QUERIES which is a mapping to the run-away log file.
+
+#### Option2: persist in kv data (chosen)
+
+Print runaway query records in tidb log file, and also persist the records in KV data by the admin table `INFORMATION_SCHEMA.RUNAWAY_QUERIES`. The logs are persisted in batch with a flush interval config `runaway_queries_history_flush_interval`. Queries on the table `INFORMATION_SCHEMA.RUNAWAY_QUERIES` would return the results combining both in-memory and on-disk data.
```
Comment (glorv) [inline docs/design/2023-06-16-runaway-queries-management.md:65]: The current implementation is different from this description

Comment (Connor1996) [reply docs/design/2023-06-16-runaway-queries-management.md:65]: updated

**Review by glorv — APPROVED** [review]

**Comment by glorv** [comment]
@nolouch PTAL

**Review by nolouch — APPROVED** [review]

**Pull request merged**